lib/core/test/unit/Cardano/Wallet/DB/SqliteSpec.hs

```diff
@@ -313,6 +381,62 @@ sqliteSpecRnd = do
             it "Sequential state machine tests"
                 (prop_sequential :: TestDBRnd -> Property)
 
+testMigrationTxMetaFee
+    :: forall k s.
+        ( s ~ SeqState 'Mainnet k
+        , k ~ ShelleyKey
+        , WalletKey k
+        , PersistState s
+        , PersistPrivateKey (k 'RootK)
+        , PaymentAddress 'Mainnet k
+        )
+    => String
+    -> Int
+    -> [(Hash "Tx", Natural)]
+    -> IO ()
+testMigrationTxMetaFee dbName expectedLength caseByCase = do
+    let orig = $(getTestData) </> dbName
+    withSystemTempDirectory "migration-db" $ \dir -> do
+        let path = dir </> "db.sqlite"
+        let ti = dummyTimeInterpreter
+        copyFile orig path
+        (logs, result) <- captureLogging $ \tr -> do
+            withDBLayer @s @k tr defaultFieldValues (Just path) ti
+                $ \(_ctx, db) -> db & \DBLayer{..} -> atomically
+                $ do
+                    [wid] <- listWallets
+                    readTxHistory wid Nothing Descending wholeRange Nothing
+
+        -- Check that we've indeed logged a needed migration for 'fee'
+        length (filter isMsgManualMigration logs) `shouldBe` 1
+
+        -- Check that the migrated history has the correct length.
+        length result `shouldBe` expectedLength
+
+        -- Verify that all incoming transactions have no fees set, and that all
+        -- outgoing ones do.
+        forM_ result $ \TransactionInfo{txInfoMeta} -> do
+            case txInfoMeta ^. #direction of
+                Incoming -> txInfoMeta ^. #fee `shouldSatisfy` isNothing
+                Outgoing -> txInfoMeta ^. #fee `shouldSatisfy` isJust
+
+        -- Also verify a few hand-picked transactions
+        forM_ caseByCase $ \(txid, expectedFee) -> do
+            case L.find ((== txid) . txInfoId) result of
+                Nothing ->
+                    fail $ "tx not found: " <> T.unpack (toText txid)
+                Just TransactionInfo{txInfoMeta} ->
+                    (txInfoMeta ^. #fee) `shouldBe` Just (Quantity expectedFee)
```

Comment:
@piotr-iohk anything else you could think of that the migrated table should satisfy. In the end, the change is only operated on the `fee` column, so there isn't much more we can test for this particular column I believe...

Reply:
Perhaps validate also the calculated deposit? It will be 0 for all cases but one in current set of cases.